packages/typespec-client-generator-core/src/package.ts

```diff
@@ -463,11 +424,18 @@ function getSdkMethods<TServiceOperation extends SdkServiceOperation>(
     const operationGroupClient = diagnostics.pipe(
       createSdkClientType<TServiceOperation>(context, operationGroup, sdkClientType),
     );
-    const clientInitialization = getClientInitialization(context, operationGroup.type);
+    const tspClientInitialization = getClientInitialization(context, operationGroup.type);
     const parameters: SdkParameter[] = [];
-    if (clientInitialization) {
-      for (const property of clientInitialization.properties) {
-        parameters.push(property);
+    if (tspClientInitialization) {
+      const clientInitialization = getSdkInitializationType(
+        context,
+        operationGroup,
+        tspClientInitialization,
+      );
+      for (const property of clientInitialization.model.properties) {
+        if (property.kind === "method") {
+          parameters.push(property);
+        }
       }
     } else {
     }
```

Comment:
if no customized client initialization, then we don't need to add parameter for client accessor?

Reply:
we still add parameters for client initialization, the api version and endpoint parameters, we just don't iterate through a model and add them